JenkinsMaster.groovy

```diff
@@ -0,0 +1,127 @@
+pipeline {
+    agent {
+        label 'docker01'
+    }
+    options {
+        ansiColor colorMapName: 'XTerm'
+    }
+    parameters {
+        string(
+            name: 'APP_SERVERS',
+            defaultValue: '10.10.10.122',
+            description: 'Deploy container to these servers. List of servers separated by comma.'
+        )
+        booleanParam(
+            name: "BUILD_IMAGE",
+            defaultValue: true,
+            description: "Build image and upload it to Docker registry"
+        )
+        string(
+            name: 'RABBIT_HOST',
+            defaultValue: '10.10.10.96',
+            description: 'RabbitMQ server'
+        )
+        string(
+            name: 'RABBIT_PORT',
+            defaultValue: '5672',
+            description: 'RabbitMQ port'
+        )
+        string(
+            name: 'GRAYLOG_HOST',
+            defaultValue: 'logs.roihunter.com',
+            description: 'Graylog server'
+        )
+        string(
+            name: 'GRAYLOG_PORT',
+            defaultValue: '12212',
+            description: 'Graylog port'
+        )
+    }
+    stages {
+        when {
+            expression {
+                return params.BUILD_IMAGE
+            }
+        }
+        stage('Build') {
+            steps {
+                sh "docker build --rm=true -t captain-hook-master ."
+            }
+        }
+
+        stage('Prepare and upload to registry ') {
+            when {
+                expression {
+                    return params.BUILD_IMAGE
+                }
+            }
+            steps {
+                withCredentials([string(credentialsId: 'docker-registry-azure', variable: 'DRpass')]) {
+                    sh 'docker login roihunter.azurecr.io -u roihunter -p "$DRpass"'
+                    sh "docker tag captain-hook-master roihunter.azurecr.io/captain-hook/master"
+                    sh "docker push roihunter.azurecr.io/captain-hook/master"
+                    sh "docker rmi captain-hook-master"
+                    sh "docker rmi roihunter.azurecr.io/captain-hook/master"
+                }
+            }
+        }
+
+        stage('Deploy API container') {
+            steps {
+                withCredentials([
+                    string(credentialsId: 'docker-registry-azure', variable: 'DRpass'),
+                    string(credentialsId: "captainhook-master-facebook-verify-token", variable: "captainhook_facebook_verify_token" ),
+                    usernamePassword(
+                        credentialsId: "captainhook-master-rabbit-userneme-password",
```

Comment:
@jimrkh There is a typo **userneme** and it is everywhere including the credentials in Jenkins :-)